Add first presubmit checks with Travis CI integration.

reillyeon opened the pull request:
This change adds a .travis.yml file instructing Travis CI to set up a Python 2.7 environment and then run the presubmit.sh script. This script runs a minimal set of PyFlakes tests through the flake8 test runner.

To ensure that the first build passes I have removed all of the unused imports in the project.

Later work on issue #135 will include enabling more of the style tests available in flake8 and adding unit tests.

reillyeon committed:
Add first presubmit checks with Travis CI integration.

This change adds a .travis.yml file instructing Travis CI to set up a
Python 2.7 environment and then run the presubmit.sh script. This script
runs a minimal set of PyFlakes tests through the flake8 test runner.

To ensure that the first build passes I have removed all of the unused
imports in the project.

Later work on issue #135 will include enabling more of the style tests
available in flake8 and adding unit tests.

kb1lqc commented:
Hey what's up with all the deletes? Thanks for this PR!

reillyeon commented:
The \_\_init\_\_.py files I deleted are for directories that are not actually used as Python modules. They contained imports that were unused.

reillyeon committed:
Remove unused variables.

This patch removes unused variables (or in the case of variables printed
for debugging purposes, hides them behind a DEBUG global) and enables
unused variable warnings in the presubmit tests.

reillyeon committed:
Fix all undefined variable warnings.

The presubmit tests can now run all PyFlakes tests.

reillyeon commented:
I've added 2 more patches to this pull request which enable the rest of the PyFlakes tests.

kb1lqc committed:
Updated Presubmit Filename

Presubmit.py was causing travis-ci to fail so trying presubmit.sh which
actually exists.

kb1lqc committed:
Update Presubmit.py Filename

Presubmit.py was causing travis-ci to fail so trying presubmit.sh which
actually exists.

kb1lqc commented on Applications/APRS/aprs.py:
```diff
@@ -12,17 +12,10 @@
 import requests
 from time import sleep
 
-# Can we clean this up?
```
Haha wow this is a good find. Totally based APRS off of Telemetry and forgot to remove the FaradayIO import code as it isn't used. @kb1lqd

kb1lqc commented on Faraday_Proxy_Tools/FaradayIO/commandmodule.py:
```diff
@@ -291,7 +290,7 @@ def packet_gpio_gps_standby_enable():
 
     .. todo:: This function needs updating to the new create_gpio_command_packet()
     """
-    return create_gpio_command_packet(3, gpio_allocation.GPS_STANDBY, 0)
```
What's up with gpio_allocations moving to gpioallocations? This seems fine it just doesn't appear to me that these two are renamed anywhere prior to this? @reillyeon @kb1lqd

kb1lqc commented on Proxy/faraday_uart_stack/layer_4_service.py:
```diff
@@ -129,7 +132,8 @@ def receive_datagram_queue_hasitem(self):
     def process_received_datagram(self, datagram):
         parsed_datagram_dict = layer_4_protocol.parse_packet(datagram)
-        #print "RX'd:", parsed_datagram_dict
+        if DEBUG:
```
Added during unused variables commit. Was this caught by the flake8 tool @reillyeon ?
https://github.com/reillyeon/Faraday-Software/commit/0ca578306425529e416344f9d280714c62c59dec

reillyeon replied:
In this case `parsed_datagram_dict` is an unused variable because the print statement was commented out. I added the `DEBUG` flag so that we could keep the print statement around but in a disabled state.

kb1lqc commented on Proxy/proxy.py:
```diff
@@ -262,7 +262,7 @@ def proxy():
             if limit is None:
                 # Optional
-                limit = len(queDict[port])
+                limit = len(getDicts[station][port])
```
@reillyeon is this you adding this or the automated tests? This was done in https://github.com/reillyeon/Faraday-Software/commit/6060044dedfafc2afb574df8bf3263e7e955c0bf where you fixed undefined variables. Is this considered one?

reillyeon replied:
`queDict` doesn't exist. It was a guess on my part that what was meant was `getDicts[station]`. Having these presubmits in place should help avoid us introducing code like this that we can't explain in the future.

kb1lqc commented on Faraday_Proxy_Tools/FaradayIO/commandmodule.py:
```diff
@@ -301,7 +300,7 @@ def packet_gpio_gps_standby_disable():
     .. todo:: This function needs updating to the new create_gpio_command_packet()
     """
-    return create_gpio_command_packet(3, gpio_allocation.GPS_STANDBY, 1)
+    return create_gpio_command_packet(3, gpioallocations.GPS_STANDBY, 1)
```
@reillyeon is this you adding this or the automated tests? This was done in https://github.com/reillyeon/Faraday-Software/commit/6060044dedfafc2afb574df8bf3263e7e955c0bf where you fixed undefined variables. Is this considered one?

reillyeon replied:
All these changes were made by hand. The module is called `gpioallocations`, I guess this code was never exercised because there is no such variable `gpio_allocations`.

kb1lqc reviewed (approved):
I added four comments to @kb1lqd  and @reillyeon. Tested with telemetry and APRS OK, Tested start 'hello world" blinking LED OK, and tested changing callsign and node ID configurations OK with hardware.

kb1lqc commented:
I added @kb1lqd onto the review but approved it. I am taking a bit of a leap of faith as I get used to this so Reilly please chime in on my comments. Overall my simplified manual testing shows what I presume to be normal operations however it is by no means exhaustive. However, skimming through the changes lots of the deletions seem to truly be unused variables and lots of the changes by Reilly were optimizations for centralizing variable definitions.

@reillyeon please let me know if you changed anything not mentioned in the commit message, maybe they are a bit vague for me to understand or I'm missing somewhere where there's more detail?

reillyeon committed:
Merge pull request #1 from kb1lqc/travis_ci

Updated Presubmit Filename

kb1lqc committed:
Merge branch 'master' into travis_ci

kb1lqc committed:
Fixed error recently commited to telemetry

During the merge of the lastest master and travis-ci branch I accidentally
committed some errors when the GitHub editor caused me to unknowingly
return a few lines erronously.

kb1lqc commented:
@reillyeon I pulled in the latest master and had to resolve a few conflicts. So far this branch seems to boot up and save telemetry to proxy using hardware just fine. However it's now failing travis-ci and the log messages are blank so I really don't have a good sense of why. I'd like to pull this in tonight!

reillyeon committed:
Remove unused dbFilename in telemetry.py.

kb1lqc commented:
Looks like it's building!
https://travis-ci.org/kb1lqc/Faraday-Software/jobs/216818128
